src/protocol/mqtt.rs

```diff
@@ -0,0 +1,123 @@
+use crate::extend::ExtendableThing;
+use serde::{Deserialize, Serialize};
+use serde_with::{serde_as, skip_serializing_none, OneOrMany};
+
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Eq, Hash)]
+pub enum ControlPacket {
+    #[serde(rename = "mqv:publish")]
+    Publish,
+    #[serde(rename = "mqv:subscribe")]
+    Subscribe,
+    #[serde(rename = "mqv:unsubscribe")]
+    Unsubscribe,
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Eq, Hash)]
+pub enum QoS {
+    #[serde(rename = "quality:0")]
+    AtMostOnce,
+    #[serde(rename = "quality:1")]
+    AtLeastOnce,
+    #[serde(rename = "quality:2")]
+    ExactlyOnce,
+}
+
+#[serde_as]
+#[skip_serializing_none]
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Eq, Hash, Default)]
+pub struct Form {
+    #[serde(
+        rename = "mqv:retain",
+        skip_serializing_if = "std::ops::Not::not",
+        default
+    )]
+    pub retain: bool,
+    #[serde(rename = "mqv:controlPacket")]
+    pub control_packet: Option<ControlPacket>,
+    #[serde(rename = "mqv:qos")]
+    pub qos: Option<QoS>,
+    #[serde(rename = "mqv:topic")]
+    pub topic: Option<String>,
```

Comment:
Mmmhh... do we need to check the validity against subscriptions? In that case we really need a callback with the whole thing or something similar...

https://w3c.github.io/wot-binding-templates/bindings/protocols/mqtt/index.html#topicname

Reply:
That is an implementation detail for `wot-serve-mqtt`, at `wot-td` level we do not have the information for that